extensions/panache/hibernate-orm-panache-kotlin/runtime/src/main/java/io/quarkus/hibernate/orm/panache/kotlin/runtime/JpaOperations.java

```diff
@@ -307,6 +330,13 @@ public static int update(Class<?> entityClass, String query, Object... params) {
             return query.stream();
         }
 
+        @Override
+        public EntityManager getEntityManager(Class<?> clazz) {
+            String clazzName = clazz.getName();
+            String persistentUnitName = entityToPersistenceUnit.get(clazzName);
+            return super.getEntityManager(persistentUnitName);
```

Comment:
I wonder if we could avoid a lot of overhead by having a generated method in the PanacheEntity/Repository return a static Literal for consumption by CDI.

It would avoid the overhead of the map lookup and would also avoid the overhead of instantiating a new Literal each time we want to the get EntityManager.

I'm not that versed into Panache so it might just be a crazy idea.

Reply:
Ah that's something I never thought of. This way seemed easier for me. I can look at that idea.

Reply:
And for the default one, we would use `javax.enterprise.inject.Default.Literal.INSTANCE`.

Reply:
Maybe let's get this first version in and then we work on a second iteration with code generation?

Reply:
Depends on how fast you will be but I think Panache generation uses ASM so it won't be fun :).

Reply:
Sure, I'll focus on resolving other comments (https://github.com/quarkusio/quarkus/pull/8695#discussion_r478451025) first.